python/mxnet/ndarray/ndarray.py

```diff
@@ -968,6 +973,14 @@ def reshape(self, shape):
         array([[-1., -1., -1.],
                [-1., -1., -1.]], dtype=float32)
         """
+        if len(shape) == 1 and isinstance(shape[0], (list, tuple)):
+            shape = shape[0]
+        elif not len(shape):
```

Comment:
what happens for reshape(1, 2, shape=1)?

Reply:
keyword argument is ignored.

Reply:
would that be a problem?

Reply:
an error should be raised

Reply:
OK

Reply:
Numpy throws an error too, though the error is confusing.
```
In [1]: import numpy as np

In [2]: a = np.ones((3,5))

In [3]: a.reshape(3,5,shape=(7,))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-cac6c36bb3d2> in <module>()
----> 1 a.reshape(3,5,shape=(7,))

TypeError: 'shape' is an invalid keyword argument for this function
```